src/components/app/details/appDetails/utils.tsx

```diff
@@ -384,127 +386,233 @@ export function addQueryParamToGrafanaURL(url: string, appId: string | number, e
+              deploymentData.deploymentStatus = DEPLOYMENT_STATUS.FAILED
               deploymentData.deploymentStatusText = 'Failed'
-              deploymentData.deploymentStatusBreakdown.APP_HEALTH.displaySubText = ': Failed'
+              deploymentData.deploymentStatusBreakdown.APP_HEALTH.displaySubText = '   Failed'
               deploymentData.deploymentError = element['statusDetail']
-          } else if (element['status'].includes('KUBECTL_APPLY')) {
+          }  else if (element['status'] === TIMELINE_STATUS.DEPLOYMENT_SUPERSEDED) {
+            deploymentData.deploymentStatus = DEPLOYMENT_STATUS.SUPERSEDED
+        } else if ( index === data.timelines.length - 1 && (element['status'] === TIMELINE_STATUS.FETCH_TIMED_OUT || element['status'] === TIMELINE_STATUS.UNABLE_TO_FETCH_STATUS)){
+            if(element['status'] === TIMELINE_STATUS.FETCH_TIMED_OUT) {
+                deploymentData.deploymentStatus = DEPLOYMENT_STATUS.TIMED_OUT
+                deploymentData.deploymentStatusText = 'Timed out'
+            } else if(element['status'] === TIMELINE_STATUS.UNABLE_TO_FETCH_STATUS) {
+                deploymentData.deploymentStatus = DEPLOYMENT_STATUS.UNABLE_TO_FETCH
+                deploymentData.deploymentStatusText = 'Unable to fetch status'
+            }
+            deploymentData.deploymentError = `Below resources did not become healthy within 10 mins. Resource status shown below was last fetched ${lastFetchedTime}. ${data.statusFetchCount} retries failed.`
+          } else if (element['status'].includes(TIMELINE_STATUS.KUBECTL_APPLY)) {
+              if(element?.resourceDetails){
+                deploymentPhases.forEach((phase) => {
+                    for(let item of element.resourceDetails){
+                        if(phase === item.resourcePhase){
+                            tableData.currentPhase = phase
+                            if(item.resourceStatus === 'failed'){
+    
+                            }
+                            tableData.currentTableData.push({icon: 'success',phase: phase, message: `${phase}: Create and update resources based on manifest` })
+                            return
+                        }
+                    }
+                  })
+              }
               if (
-                  element['status'] === 'KUBECTL_APPLY_STARTED' &&
+                  element['status'] === TIMELINE_STATUS.KUBECTL_APPLY_STARTED &&
                   deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.time === '' &&
-                  deploymentData.deploymentStatus !== 'Succeeded'
+                  deploymentData.deploymentStatus !== DEPLOYMENT_STATUS.SUCCEEDED
               ) {
-                  if (deploymentData.deploymentStatus === 'failed') {
+                deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.resourceDetails = element.resourceDetails.filter((item) => item.resourcePhase ===  tableData.currentPhase)//// check this condition
```

Comment:
Is this comment valid? If yes then can we address the same?

Reply:
done